packages/qdrant-loader/src/qdrant_loader/core/pipeline/document_pipeline.py

```
@@ -46,11 +51,129 @@ def __init__(
+                            await graph_store.upsert_nodes_batch(nodes_batch)
+                        if edges_batch:
+                            await graph_store.upsert_edges_batch(edges_batch)
+                        logger.info("🔄 Add graph store successfully...")
+                    except Exception as e:
+                        logger.error(
+                            "⚠️ Graph upsert failed (non-fatal): %s",
+                            e,
+                            exc_info=True,
+                        )
+                logger.info(
+                    f"Graph batch size: nodes={len(nodes_batch)}, edges={len(edges_batch)}"
```

Ruff (0.15.14)

[warning] Logging statement uses f-string

(G004)